nautobot/extras/management/__init__.py

```diff
@@ -77,14 +81,213 @@
+
+    choices = []
+
+    for _, value in choiceset.CHOICES:
+        choice_kwargs = {
+            "name": value,
+            "description": description_map[value],
+            "color": color_map[value],
+        }
+        choices.append(choice_kwargs)
+
+    return choices
+
+
+def create_custom_roles(
+    verbosity=2,
+    apps=global_apps,
+    models=None,
+    **kwargs,
+):
+    """
+    Create database Role choices from choiceset enums.
+
+    This is called during data migrations for importing `Role` objects from
+    `ChoiceSet` enums in flat files.
+    """
+
+    # Only print a newline if we have verbosity!
+    if verbosity > 0:
+        print("\n", end="")
+
+    if "test" in sys.argv:
+        # Do not print output during unit testing migrations
+        verbosity = 1
+
+    if not models:
+        models = ROLE_CHOICESET_MAP.keys()
+
+    # Prep the app and get the Role model dynamically
+    try:
+        Role = apps.get_model("extras.Role")
+        ContentType = apps.get_model("contenttypes.ContentType")
+    except LookupError:
+        return
+
+    added_total = 0
+    linked_total = 0
+
+    # Iterate choiceset kwargs to create role objects if they don't exist
+    for model_path in models:
+        choiceset = ROLE_CHOICESET_MAP[model_path]
+        content_type = ContentType.objects.get_for_model(apps.get_model(model_path))
+        choices = export_roles_from_choiceset(choiceset)
+
+        if verbosity >= 2:
+            print(f"    Model {model_path}", flush=True)
+
+        for choice_kwargs in choices:
+            defaults = choice_kwargs.copy()
+            name = defaults.pop("name")
+            try:
+                obj, created = Role.objects.get_or_create(name=name, defaults=defaults)
+            except IntegrityError as err:
+                raise SystemExit(
+                    f"Unexpected error while running data migration to populate role for {model_path}: {err}"
+                ) from err
+
+            # Make sure the content-type is associated.
+            if content_type not in obj.content_types.all():
+                obj.content_types.add(content_type)
+
+            if created and verbosity >= 2:
+                print(f"      Adding and linking role {obj.name}", flush=True)
+                added_total += 1
+            elif not created and verbosity >= 2:
+                print(f"      Linking to existing role {obj.name}", flush=True)
+                linked_total += 1
+
+    if verbosity >= 2:
+        print(f"    Added {added_total}, linked {linked_total} role records")
+
+
+def clear_role_choices(
+    apps=global_apps,
+    verbosity=2,
+    models=None,
+    clear_all_model_roles=True,
+    **kwargs,
+):
+    """
+    Remove content types from roles, and if no content types remain, delete the roles as well.
+    """
+    if "test" in sys.argv:
+        # Do not print output during unit testing migrations
+        verbosity = 1
+
+    if verbosity >= 0:
```

Comment:
Doesn't it seem like this should be:

```suggestion
        verbosity = 0

    if verbosity > 0:
```